src/SWP/Bundle/ContentBundle/Loader/ArticleLoader.php

```diff
@@ -78,7 +80,7 @@ public function load($type, $parameters, $responseType = LoaderInterface::SINGLE
                     ->findOneBy(['slug' => $parameters['slug']]);
             }
 
-            if (!is_null($article)) {
+            if (!is_null($article) && $publishWorkflowChecker->isGranted(PublishWorkflowChecker::VIEW_ATTRIBUTE, $article)) {
```

Comment:
What is `getMeta`? Meta class is unaware about type of object pushed there. This kind of logic shouldn't be there but in loaders. And it's up to loader how he want to work with access permissions.